cluster/get-kube.sh

```diff
@@ -63,7 +63,31 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release/release}"
+# If KUBERNETES_RELEASE_URL is overridden but KUBERNETES_CI_RELEASE_URL is not then set KUBERNETES_CI_RELEASE_URL to KUBERNETES_RELEASE_URL.
+KUBERNETES_CI_RELEASE_URL="${KUBERNETES_CI_RELEASE_URL:-${KUBERNETES_RELEASE_URL:-https://dl.k8s.io/ci}}"
+KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://dl.k8s.io}"
+
+KUBE_RELEASE_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha)\\.(0|[1-9][0-9]*))?$"
+KUBE_CI_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-(beta|alpha)\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*)\\+[-0-9a-z]*)?$"
+
+# Sets KUBE_VERSION variable if an explicit version number was provided (e.g. "v1.0.6",
+# "v1.2.0-alpha.1.881+376438b69c7612") or resolves the "published" version
+# <path>/<version> (e.g. "release/stable",' "ci/latest-1") by reading from GCS.
+#
+# See the docs on getting builds for more information about version
+# publication.
+#
+# Args:
+#   $1 version string from command line
+# Vars set:
+#   KUBE_VERSION
+function set_binary_version() {
+  if [[ "${1}" =~ "/" ]]; then
+    KUBE_VERSION=$(curl -fL "https://dl.k8s.io/${1}.txt")
```

Comment:
I finally tested this script myself, and now realize we really want `-sS` here too, so we don't get the progress output (which is unnecessary).

may also be a good idea to improve the error message. e.g.
```patch
-    KUBE_VERSION=$(curl -fL "https://dl.k8s.io/${1}.txt")
+    KUBE_VERSION=$(curl -fsSL "https://dl.k8s.io/${1}.txt") || {
+      echo "Unknown version '$1'" >&2; exit 1; }
```
then becomes
```shell
$ KUBERNETES_RELEASE=bad/version cluster/get-kube.sh
curl: (22) The requested URL returned error: 404 Not Found
Unknown version 'bad/version'
```